user/service.py

```diff
@@ -30,17 +38,18 @@ def create(username: str, password: str) -> Optional[str]:
 
 def update(user_id: str, username: str, password: str):
     """Update a user given its ID."""
-    user = user_dao.get_by_id(user_id)
-
+    user = None
+    try:
+        user = get(user_id)
+    except validation_exceptions.ValidationException as e:
+        raise e
     salt = user.password_salt
     user_dao.update(user_id, username, hash_password(password + salt))
 
 def delete(user_id: str):
+    get(user_id) # raises, which can be caught by controller
     """Delete a user given its ID."""
-    if user_dao.get_by_id(user_id) is None:
-        return False
-
-    return user_dao.delete(user_id)
+    user_dao.delete(user_id)
```

Comment:
One step further, since the DAO checks this condition (via `get_by_id`) too this can be simplified to just
```python
def delete(user_id: str):
    user_dao.delete(user_id)
```